scripts/apps/authoring/authoring/directives/AuthoringDirective.js

```diff
@@ -294,57 +331,96 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
                 return true;
             }
 
+            /**
+             * Checks if associations is with rewrite_of item then open then modal to add associations.
+             * The user has options to add associated media to the current item and review the media change
+             * or publish the current item without media.
+             */
+            function checkMediaAssociatedToUpdate() {
+                let rewriteOf = $scope.item.rewrite_of;
+
+                if (!config.features.editFeaturedImage ||
+                    $scope.action === 'kill' || !rewriteOf ||
+                    $scope.item.associations && $scope.item.associations.featuremedia) {
+                    return $q.when(true);
+                }
+
+                return api.find('archive', rewriteOf)
+                    .then((rewriteOfItem) => {
+                        if (rewriteOfItem && rewriteOfItem.associations &&
+                            rewriteOfItem.associations.featuremedia) {
+                            return confirm.confirmFeatureMedia(rewriteOfItem);
+                        }
+                        return true;
+                    })
+                    .then((result) => {
+                        if (result && result.associations) {
+                            $scope.item.associations = result.associations;
+                            $scope.autosave($scope.item);
+                            return false;
+                        }
+
+                        return true;
+                    });
+            }
+
             function publishItem(orig, item) {
                 var action = $scope.action === 'edit' ? 'publish' : $scope.action;
 
                 validate(orig, item);
 
-                return authoring.publish(orig, item, action)
-                .then((response) => {
-                    if (!response) {
-                        notify.error(gettext('Unknown Error: Item not published.'));
-                        return false;
-                    }
+                return checkMediaAssociatedToUpdate()
+                    .then((result) => {
+                        if (result) {
+                            return authoring.publish(orig, item, action);
+                        }
+                        return $q.reject(false);
+                    })
+                    .then((response) => {
+                        if (!response) {
+                            notify.error(gettext('Unknown Error: Item not published.'));
+                            return false;
```

Comment:
I think it would be better to `reject` here so that chain will stop (in case there is another `then` added later)